### ISSUE TITLE
@Parameter info are ignored for header 

### ISSUE THREAD

**Issue by btocquec** [springdoc/springdoc-openapi#544]
Hi,

Here is my controller part:
```
...
    @Operation(summary = "find-articles")
    @Parameter(name = HttpHeaders.IF_MODIFIED_SINCE,
            description = "DateTime",
            in = ParameterIn.HEADER,
            schema = @Schema(type = "string", format = "date-time"),
            example = "2020-01-01T00:00:00.000Z"
    )
    public ResponseEntity<Page<Article>> findArticles(
            @RequestHeader(value = HttpHeaders.IF_MODIFIED_SINCE, required = false) Instant modifiedSince, ...
```

I expect to see my **example** and my **description** in it but I've only got:
![image](https://user-images.githubusercontent.com/8244285/78684873-4a773800-78f1-11ea-8693-ce195ef0ad71.png)

If I put this part under "parameters" in @Operation it works but as I want to create my own reusable annotation, I'd like this @Parameter to be used.

Looking deeper in the code, it appears that the binding can't be done between `modifiedSince` and `HttpHeaders.IF_MODIFIED_SINCE` where as the merge works if in @Operation.

Regards,

**Comment by bnasslahsen** [comment]
@btocquec, 

 The merge doesn't happen, because your parameter name, doesn't have the same, as the one used in the swagger annotation `@Parameter`.
 Here the working example:

```java
	@Operation(summary = "find-articles")
	@GetMapping
	@Parameter(name = "modifiedSince",
			description = "DateTime",
			in = ParameterIn.HEADER,
			schema = @Schema(type = "string", format = "date-time"),
			example = "2020-01-01T00:00:00.000Z"
	)
	public ResponseEntity<String> findArticles(@RequestHeader(value = HttpHeaders.IF_MODIFIED_SINCE, required = false) Instant modifiedSince) {
		return null;
	}

```

**Comment by btocquec** [comment]
Yes but no :)
In that case, the parameter filled in the UI is not detected by my webService => the `modifiedSince` is null !

![image](https://user-images.githubusercontent.com/8244285/78705285-8ddf9f80-790d-11ea-8985-35ef4a13fc37.png)

**Comment by bnasslahsen** [comment]
You have a first workaround, is to pass the

**Comment by btocquec** [comment]
Yes, I've already noticed this one and thanks for that but this my aim is to:

- have a reusable annotation that I can just declare on each needed method
- less "intrusive" code (prefer separating rest part and swagger context)

Exact, I have no issue for others parameters (path param for example) but for Header, the name of the java parameter and the name of the http parameter are different that's why I guess wwe can consider this as an issue?

Regards,

**Comment by bnasslahsen** [comment]
@btocquec, 

 Yeah, of course and the fix the should be available for the next release: v1.3.2

**Comment by btocquec** [comment]
Thanks, Great Job !!
I'll confirm/retest with next release.
